### PR TITLE
feat: make PhiAccrualFailureDetector configurable 

### PR DESCRIPTION
**Pull request by tisonkun**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

This closes https://github.com/GreptimeTeam/greptimedb/issues/1738.

Although, `PhiAccrualFailureDetector` seems a deep internal struct. Perhaps we can evaluate if we want to config it in metasrv level.

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

### CODE REVIEW

**Comment by tisonkun** [comment]
@MichaelScofield @WenyXu @killme2008

File: tests/conf/datanode-test.toml.template [hunk 1]
```diff
@@ -22,6 +22,12 @@ timeout_millis = 3000
 connect_timeout_millis = 5000
 tcp_nodelay = false
 
+[failure_detector]
```
Comment (tisonkun) [inline tests/conf/datanode-test.toml.template:25]: I'm not sure if we should add this section to `standalone.toml.template` and `metasrv.toml.template` also.

File: src/meta-srv/src/failure_detector.rs [hunk 2]
```diff
@@ -67,13 +77,28 @@ impl Default for PhiAccrualFailureDetector {
...
             first_heartbeat_estimate_millis: 1000,
-            heartbeat_history: HeartbeatHistory::new(1000),
-            last_heartbeat_millis: None,
         }
     }
 }
 
+impl Default for PhiAccrualFailureDetector {
+    fn default() -> Self {
+        PhiAccrualFailureDetector::from_options(PhiAccrualFailureDetectorOptions::default())
```
Comment (tisonkun) [inline src/meta-srv/src/failure_detector.rs:86]: Perhaps `Self::from_options(Default::default())`. I don't have a strong preference and don't know the style preferred in greptimedb.

File: src/meta-srv/src/failure_detector.rs [hunk 3]
```diff
@@ -14,6 +14,8 @@
 
 use std::collections::VecDeque;
 
+use serde::{Deserialize, Serialize};
+
 /// This is our port of Akka's "[PhiAccrualFailureDetector](https://github.com/akka/akka/blob/main/akka-remote/src/main/scala/akka/remote/PhiAccrualFailureDetector.scala)"
```
Comment (tisonkun) [inline src/meta-srv/src/failure_detector.rs:19]: BTW, Akka has switched to BuSL 1.1 which is not an open-source license. And code translation is still under the rick of being a derivative work.

So either specify the tag of version still released under Apache License 2.0, or try to use a link to https://github.com/apache/incubator-pekko/blob/main/remote/src/main/scala/org/apache/pekko/remote/PhiAccrualFailureDetector.scala.

Comment (MichaelScofield) [reply src/meta-srv/src/failure_detector.rs:19]: That's great to know, thank you for pointing it out. Would you kind enough to change it in this pr? We can link to the code in akka version 2.6:
https://github.com/akka/akka/blob/v2.6.21/akka-remote/src/main/scala/akka/remote/PhiAccrualFailureDetector.scala

File: src/meta-srv/src/failure_detector.rs (hunk 2 above)
Comment (MichaelScofield) [reply src/meta-srv/src/failure_detector.rs:87]: Both are fine, we don't have a style guide on this either. Though I personally prefer `Self`.

File: src/meta-srv/src/failure_detector.rs (hunk 3 above)
Comment (tisonkun) [reply src/meta-srv/src/failure_detector.rs:19]: OK. Let's do it in another PR. I can preapre one of course.

File: src/meta-srv/src/failure_detector.rs (hunk 2 above)
Comment (tisonkun) [reply src/meta-srv/src/failure_detector.rs:87]:
```suggestion
        Self::from_options(Default::default())
```

Yes. I notice that `Self` is preferred in most of this codebase.

**Comment by tisonkun** [comment]
@MichaelScofield Is there any pending task before this PR mergeable?

File: tests/conf/datanode-test.toml.template (hunk 1 above)
Comment (MichaelScofield) [reply tests/conf/datanode-test.toml.template:25]: I think this section shouldn't be added here. The `MetaSrvOptions` (which contains the `PhiAccrualFailureDetectorOptions`) is created in method `load_options` in `cmd/src/metasrv.rs`, which loads the config files provided by the user. So we can add this section in `config/metasrv.example.toml`. Also you can change the test `test_read_from_config_file` for this new section.

**Comment by MichaelScofield** [comment]
Left a comment, PTAL @tisonkun

**Comment by tisonkun** [comment]
@MichaelScofield Updated.

**Review by MichaelScofield — APPROVED** [review]

**Review by fengjiachun — APPROVED** [review]
LGTM

File: src/meta-srv/src/failure_detector.rs (hunk 3 above)
Comment (tisonkun) [reply src/meta-srv/src/failure_detector.rs:19]: Here it is - https://github.com/GreptimeTeam/greptimedb/pull/2714

File: config/metasrv.example.toml [hunk 4]
```diff
@@ -28,6 +28,13 @@ max_retry_times = 12
 # Initial retry delay of procedures, increases exponentially
 retry_delay = "500ms"
 
+# Failure detectors options.
+[failure_detector]
+threshold = 8.0
+min_std_deviation_millis = 100.0
```
Comment (killme2008) [inline config/metasrv.example.toml:34]: I think we should not use the `xxx_millis` such names for parameters. Can we use `humantime` just like other parameters for time settings? Can you create another PR to fix them? Thanks a lot. @tisonkun 

Please note such changes when doing the code review .@fengjiachun @MichaelScofield

**Pull request merged**